src/components/side_panel/find_and_replace/find_and_replace.ts

```diff
@@ -72,12 +81,58 @@ export class FindAndReplacePanel extends Component<Props, SpreadsheetChildEnv> {
 
     onMounted(() => this.focusInput());
 
+    useEffect(
+      () => {
+        this.focusButton(this.lastClickedButton);
+      },
+      () => [this.env.model.getters.getActiveSheetId()]
+    );
+
+    useEffect(
+      () => {
+        if (
+          this.state.searchOptions.searchScope === "specificRange" &&
+          !this.state.searchOptions.specificRange
+        ) {
+          this.state.searchOptions.specificRange = this.lastSpecificRange;
+        } else if (
+          this.state.searchOptions.searchScope !== "specificRange" &&
+          this.state.searchOptions.specificRange
+        ) {
+          this.lastSpecificRange = this.state.searchOptions.specificRange;
+          this.state.searchOptions.specificRange = undefined;
+        }
+        this.updateSearch();
+      },
+      () => [this.state.searchOptions.searchScope]
+    );
+
     onWillUnmount(() => {
       this.env.model.dispatch("CLEAR_SEARCH");
       this.env.model.dispatch("SET_FORMULA_VISIBILITY", { show: this.showFormulaState });
     });
   }
 
+  private focusButton(buttonName: string) {
+    setTimeout(() => {
+      const el = this.selectionInputref.el!;
+      if (el) {
+        const buttonElement = this.findAndReplaceRef.el!.querySelector(
+          `button[name=${buttonName}]`
+        ) as HTMLButtonElement;
+        buttonElement.focus();
+      }
+    }, 0);
+  }
+
+  onFocusSearch() {
+    const el = this.selectionInputref.el!;
+    if (el) {
+      const confirmButton = el.querySelector(".o-btn-action.o-selection-ok") as HTMLButtonElement;
+      confirmButton.click();
+    }
```

Comment:
If you need to do `querySelector` to select something on a child component, your code is most likely wrong. Can't you mimick the behaviour of the confirm button of the input, ie. dispatch  a `UNFOCUS_SELECTION_INPUT`. I think it will cancel the selection rather than conforming it, but it's not that bad. And functionnaly its better to call this on a click anywhere on the side pannel rather than on focus on the search input.

 Or probably better, and I think I told you to do that, SelectionInput could call it's confirm on `blur`. Probably with a new props `confirmOnBlur :boolean`.